### ISSUE TITLE
PHP installation via `pie` command

### ISSUE THREAD

**Issue by kynx** [grpc/grpc#38209]
### Is your feature request related to a problem? Please describe.
Support installation of PHP extension via `pie` command

### Describe the solution you'd like
PIE is the new PHP extension installer - a replacement for the old `pecl` command. See https://github.com/php/pie. It'd be great if the gprc extension could be installed with it.

### Additional context
See https://github.com/php/pie/blob/main/docs/extension-maintainers.md for the steps to make this happen. Basically it's:

1. Create a `composer.json` file in the root of this repository.
2. Register it on packagist.org

I'm happy to do a PR, but I see there is already a `composer.json` file in the root of this repository. Is this still used for anything?

**Comment by vencakrecl** [comment]
@stanley-cheung version 1.0.0 has been officially released https://github.com/php/pie/releases/tag/1.0.0

**Comment by ruudk** [comment]
@vencakrecl Could you maybe create the PR?

**Comment by vencakrecl** [comment]
@ruudk Yes, I could. I need to understand the release process of the package first. The PIE needs to have GitHub Actions and composer.json for the extension. I've started making changes to the PHP package to know how the release process works, see https://github.com/grpc/grpc/issues/40192

**Comment by AlexanderOMara** [comment]
Note: [Since September](https://web.archive.org/web/20251006181817/https://pecl.php.net/), PECL is officially deprecated in favor of PIE.

> **Have you heard about PIE?**
>
> PECL is now deprecated. 🥧 PHP Installer for Extensions (PIE) is the replacement for PECL. We recommend publishing your extension using PIE instead. 
>
> **Find out more at [https://github.com/php/pie](https://web.archive.org/web/20251217014807/https://github.com/php/pie).**

**Comment by vencakrecl** [comment]
base on this discussion https://github.com/protocolbuffers/protobuf/pull/24694, the best approach is to mirror the composer package and the extension package because you need to have composer.json in the root of repo.
* php package - https://github.com/grpc/grpc-php
* php extension package - https://github.com/grpc/grpc-php-ext

we can use https://github.com/danharrin/monorepo-split-github-action

Can you help @pawbhard?

**Comment by BSN4** [comment]
For anyone blocked on this — we built [grpc-php-rs](https://github.com/BSN4/grpc-php-rs), a drop-in replacement for ext-grpc written in Rust. It's installable via PIE today:

```
pie install bsn4/grpc
```

Pre-built binaries for PHP 8.2–8.5, Linux (x86_64/ARM64, NTS/ZTS) and macOS ARM64. No compilation needed.

It uses tonic for gRPC and rustls for TLS, which also solves the long-standing ZTS/TSRM crashes and OpenSSL/BoringSSL conflicts with ext-curl.

All existing `Grpc\` namespace code works unchanged — same API surface as ext-grpc.

**Comment by lcobucci** [comment]
@pawbhard I'd recommend you closing this, given the mirror on https://github.com/grpc/grpc-php-ext is working quite well (thank you and @bshaffer on https://github.com/grpc/grpc/pull/42784 for that, btw 👍).

The only missing bit is updating docs (https://github.com/grpc/grpc#to-start-using-grpc still mentions `pecl`).
